halibot/halmodule.py

```diff
@@ -4,18 +4,23 @@
 class HalModule(HalObject):
 
 	def reply(self, msg0=None, **kwargs):
+		# Create the reply message
 		body = kwargs.get('body', msg0.body)
 		mtype = kwargs.get('type', msg0.type)
 		author = kwargs.get('author', msg0.author)
 		origin = kwargs.get('origin', msg0.origin)
 
-		msg = Message(body=body, type=mtype, author=author, origin=origin)
-
 		# For deprecation, remove for 1.0
 		if 'context' in kwargs:
 			msg.context = kwargs['context']
 
-		self.send_to(msg, [ msg.origin ])
+		msg = Message(body=body, type=mtype, author=author, origin=origin)
+
+		# Synchronous reply?
+		if msg0.sync:
```

Comment:
Where is `msg0` set? Automatically in a `sync_` environment? Or does this need to be provided by the module implementation of `.receive()`?